server/app/models/stack_deploy.rb

```diff
@@ -1,9 +1,12 @@
 class StackDeploy
   include Mongoid::Document
   include Mongoid::Timestamps
+  include Mongoid::Enum
 
   index({ stack_id: 1 })
 
+  enum :deploy_state, [:created, :ongoing, :success, :error], default: :created
+
```

Comment:
Add also a `reason`field as the CLI is expecting that on `wait_deploy_to_finish`